backend/battle/views.py

```diff
@@ -74,8 +75,12 @@ class BattleDetail(DetailView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        team = Team.objects.filter(battle=self.object, trainer=self.request.user)
-        context['team'] = team
+        team_user = Team.objects.filter(battle=self.object, trainer=self.request.user)
```

Comment:
Will the user always be the creator or opponent? If so, you're making an unncessary query here.
You can instead move this two lines below and do:
```python
team_user = team_creator if self.request.user == self.object.creator else team_opponent
```

Reply:
This way is easier to understand, thanks!

Reply:
I tried this right now and I realized the team_creator and team_opponent are not a user object, they are an object with pokemons related, so if I do this logic, the code breaks

Reply:
The only thing you do with `team_user` is checking if it exists in the template.
But you already check if the team exists inside `get_pokemons_team`. So you're repeating a query without need.

**I think if you check if self.request.user is either the creator or opponent, and also check if one of pokemons_creator and pokemons_opponent is empty, you get the same result.**

So you can switch `team_user` to `user_has_team` and do: `user_has_team = self.request.user in [self.object.creator, self.object.opponent] and (not pokemons_creator or not pokemons_opponent)`

Please double check and lmk if you have questions.

Reply:
Thanks, I put this code and I tested it, it didn't work well, because for example, when the -- user logged -- is the creator and the user has not a team yet, but the opponent does,  "user_has_team" turns to True, instead of False. Therefore I change the old logic and deleted the unnecessary query. Let me know if you think it's clear to understand the logic. Thanks again!